modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/kill/KillCommandHandler.java

```diff
@@ -102,7 +102,7 @@ public CompletableFuture<Boolean> handle(KillCommand cmd) {
 
         CompletableFuture<Boolean> killFut = invokeCancel(handler, cmd.operationId());
 
-        if (killFut.isDone() || !cmd.noWait()) {
+        if (killFut.isCompletedExceptionally() || !cmd.noWait()) {
```

Comment:
I miss smth here, if noWait == true - we must return immediatelly i.e. return CompletableFuture.completedFuture(true); otherwise need to return killFut, isn`t it ?

Reply:
Yes

> if noWait == true - we must return immediatelly i.e. return CompletableFuture.completedFuture(true);

...except handling exception described in javadoc

`@throws IllegalArgumentException If the operation identifier is not in the correct format.`

For example ``KILL QUERY '123' NO WAIT` throws exception instead of returning true :thinking: